File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        hashSet = set()

        for num in nums:
            hashSet.add(num)

        res = 0
        for i in range(len(nums)):
            num = nums[i]
            if not num-1 in hashSet:
                length = 0
                while num in hashSet:
                    length += 1
                    num += 1
                res = max(length, res)

        return res
        

File: Data Structures & Algorithms/house-robber/submission-2.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int rob1 = 0;
        int rob2 = nums[0];

        for (int i = 1; i < nums.size(); i++) {
            int temp = rob2;
            rob2 = max(nums[i] + rob1, rob2);
            rob1 = temp;
        }

        return max(rob1, rob2);
    }
};
